app-android/src/androidTest/java/com/ustadmobile/port/android/view/WebChunkFragmentTest.kt

```diff
@@ -0,0 +1,136 @@
+
+    @JvmField
+    @Rule
+    val screenRecordRule = AdbScreenRecordRule()
+
+    @JvmField
+    @Rule
+    val crudIdlingResourceRule = ScenarioIdlingResourceRule(CrudIdlingResource())
+
+    @get:Rule
+    var permissionRule = GrantPermissionRule.grant(Manifest.permission.WRITE_EXTERNAL_STORAGE,
+            Manifest.permission.READ_EXTERNAL_STORAGE, Manifest.permission.ACCESS_COARSE_LOCATION)
+
+    var container: Container? = null
+
+    @Before
+    fun setup() {
+
+        val tmpDir = UmFileUtilSe.makeTempDir("testWebChunk",
+                "" + System.currentTimeMillis())
+
+        val chunkCountingOut = File(tmpDir, "counting-out.zip")
+
+        copyInputStreamToFile(
+                javaClass.getResourceAsStream("/com/ustadmobile/app/android/counting-out-1-20-objects.zip"),
+                chunkCountingOut)
+
+        val targetEntry = ContentEntry()
+        targetEntry.title = "tiempo de prueba"
+        targetEntry.thumbnailUrl = "https://www.africanstorybook.org/img/asb120.png"
+        targetEntry.description = "todo el contenido"
+        targetEntry.publisher = "CK12"
+        targetEntry.author = "borrachera"
+        targetEntry.primaryLanguageUid = 53
+        targetEntry.leaf = true
+        targetEntry.contentEntryUid = dbRule.db.contentEntryDao.insert(targetEntry)
+
+        container = Container()
+        container?.mimeType = "application/webchunk+zip"
+        container?.containerContentEntryUid = targetEntry.contentEntryUid
+        container?.containerUid = dbRule.db.containerDao.insert(container!!)
+
+        val containerManager = ContainerManager(container!!, dbRule.db, dbRule.repo,
+                tmpDir.absolutePath)
+        addEntriesFromZipToContainer(chunkCountingOut.absolutePath, containerManager)
+
+    }
+
+    @AdbScreenRecord("given contentEntry when web view loads then show web chunk")
+    @Test
+    fun givenContentEntry_whenWebViewLoads_thenShowWebChunk() {
+
+        val fragmentScenario = launchFragmentInContainer(themeResId = R.style.UmTheme_App,
+                fragmentArgs = bundleOf(UstadView.ARG_CONTENT_ENTRY_UID to container!!.containerContentEntryUid, UstadView.ARG_CONTAINER_UID to container!!.containerUid)) {
+            WebChunkFragment().also {
+                it.installNavController(systemImplNavRule.navController)
+            }
+        }.withScenarioIdlingResourceRule(dataBindingIdlingResourceRule)
+                .withScenarioIdlingResourceRule(crudIdlingResourceRule)
+
+
+        var count = 0
+        repeat(5) {
+            try {
+                onWebView(allOf(isDisplayed(), isJavascriptEnabled()))
+                        .withElement(findElement(Locator.CSS_SELECTOR,
+                                "div[data-test-id=tutorial-page]"))
+            } catch (io: RuntimeException) {
+                Thread.sleep(2000)
+                count++
+            }
+        }
+        
+        if(count == 5){
```

Comment:
For clarity I would define a timeout and a wait time as constant objects. I would then put the re-throw inside the catch clause.